clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1625,6 +1625,25 @@ public void testHeartbeatThreadClose() throws Exception {
             assertFalse("Heartbeat thread active after close", threads[i].getName().contains(groupId));
     }
 
+    @Test
+    public void testAutoCommitAfterCoordinatorBackToService() {
+        ConsumerCoordinator coordinator = buildCoordinator(new Metrics(), assignors,
+                ConsumerConfig.DEFAULT_EXCLUDE_INTERNAL_TOPICS, true, true);
+        subscriptions.assignFromUser(Collections.singleton(t1p));
+        subscriptions.committed(t1p, new OffsetAndMetadata(0));
+        subscriptions.seek(t1p, 100L);
+
+        coordinator.coordinatorDead();
+        assertTrue(coordinator.coordinatorUnknown());
+        client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        client.prepareResponse(offsetCommitResponse(Collections.singletonMap(t1p, Errors.NONE)));
+
+        // Async commit offset should find coordinator
+        coordinator.maybeAutoCommitOffsetsAsync(time.nanoseconds() + autoCommitIntervalMs); //make sure it does happen
```

Comment:
I think `nanoseconds` should be `milliseconds`. It's a little more idiomatic to do it like this:

```java
time.sleep(autoCommitIntervalMs);        
coordinator.maybeAutoCommitOffsetsAsync(time.milliseconds());
```